### ISSUE TITLE
[BUG] Additional nodes configured with incorrect CA certificate

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1413]
0.3.0

**To Reproduce**
1. Create 3-node Harvester cluster
2. Add a 4th node

**Expected behavior**
/etc/rancher/rke2/rke2.yaml from 4th node can be used to interact with the K8s API

**Actual behavior**

`/etc/rancher/rke2/rke2.yaml` contains a CA cert that is different from the other 3 nodes. 
Using this kubeconfig results in errors.

`/etc/rancher/rke2/rke2.yaml` on node1:

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJlVENDQVIrZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWtNU0l3SUFZRFZRUUREQmx5YTJVeUxYTmwKY25abGNpMWpZVUF4TmpNek9UUTVNamM1TUI0WERUSXhNVEF4TVRFd05EYzFPVm9YRFRNeE1UQXdPVEV3TkRjMQpPVm93SkRFaU1DQUdBMVVFQXd3WmNtdGxNaTF6WlhKMlpYSXRZMkZBTVRZek16azBPVEkzT1RCWk1CTUdCeXFHClNNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJOcS8vb0xEcHRKcUJhV3c1Y1hVOEd3T1FQbXZhbnJ2czhkRUxuTGwKaytlbWh0UlZjdlZ4SkJOY0NBME9aSEV1QTRvRy9xemRib1lDaHM4ZWc3WVpDMENqUWpCQU1BNEdBMVVkRHdFQgovd1FFQXdJQ3BEQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01CMEdBMVVkRGdRV0JCUzRWcDREQXFoR3V4RTZTNS9pCmlneDE1RFFCMmpBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlBSUprZmxFNFhHdDBVL3o0eHk3SmZoQUxrT01Ea2cKcnBFckR1QU9ZQ1ZnQVFJaEFKT3QrMU1GRGhuNGF6Q3BBUzVkblNwb3B3dFI5cmUxMU9HaFZ4KzJhVFRxCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
    server: https://127.0.0.1:6443
  name: default
contexts:
- context:
    cluster: default
    user: default
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: default
  user:
    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJrVENDQVRpZ0F3SUJBZ0lJSWFnbHlFa2ZtaEl3Q2dZSUtvWkl6ajBFQXdJd0pERWlNQ0FHQTFVRUF3d1oKY210bE1pMWpiR2xsYm5RdFkyRkFNVFl6TXprME9USTNPVEFlRncweU1URXdNVEV4TURRM05UbGFGdzB5TWpFdwpNVEV4TURRM05UbGFNREF4RnpBVkJnTlZCQW9URG5ONWMzUmxiVHB0WVhOMFpYSnpNUlV3RXdZRFZRUURFd3h6CmVYTjBaVzA2WVdSdGFXNHdXVEFUQmdjcWhrak9QUUlCQmdncWhrak9QUU1CQndOQ0FBUk5GSmdiOFpKNVBTaTMKUExVem50bG9McGI4TmQyTm5UOUdDYTJGQyt3OXlSWllnQ0lDRjQxZTBWdXBzaTNGcFREejBqSFRodHNaSEc4NgpLYldmRWFFTm8wZ3dSakFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJS3dZQkJRVUhBd0l3Ckh3WURWUjBqQkJnd0ZvQVVYUWdiNFJNcy9TcUxnc1k3aUd2QU9ndjM3UWd3Q2dZSUtvWkl6ajBFQXdJRFJ3QXcKUkFJZ1d4OXBTc1JXSEJyZ1c0OXlxbzRmeGF6dWVXSGRNWTdEWkl3a2JGckE5UmdDSUd6Z1FIM0lFbXk5NzIxaQpVbzI4L0JWc0x3Qy9ycStnZDR0TnM1Rk9CSGUzCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJlVENDQVIrZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWtNU0l3SUFZRFZRUUREQmx5YTJVeUxXTnMKYVdWdWRDMWpZVUF4TmpNek9UUTVNamM1TUI0WERUSXhNVEF4TVRFd05EYzFPVm9YRFRNeE1UQXdPVEV3TkRjMQpPVm93SkRFaU1DQUdBMVVFQXd3WmNtdGxNaTFqYkdsbGJuUXRZMkZBTVRZek16azBPVEkzT1RCWk1CTUdCeXFHClNNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJMZ0xxdGNjRHNYbjBNSUs3S1V2bWw1S1FTZVN6KzBobWhCaDdCNUoKcUdtdGp4ZVdKNy9Da045WVZMSEo0dG40Q0dmTmFLVnJOdkcxVkhWSnJYdmNhVCtqUWpCQU1BNEdBMVVkRHdFQgovd1FFQXdJQ3BEQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01CMEdBMVVkRGdRV0JCUmRDQnZoRXl6OUtvdUN4anVJCmE4QTZDL2Z0Q0RBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlBNTc5dmlLdlExYzh3Tk85L1lZeFhuVjdkeG91TGwKeXlQSERLUzMwMDFkUmdJaEFQUmJqcXBGaE91azFCOGJJdHZIb3VJWThsbkhsY29ZZGFjZDh2VkQ1bHl1Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
    client-key-data: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSVBVNkNUMFV5YlFXUGhMSTBVNzdrMkRKYUFPc0FPQ3REQkd5OXJ1aEQ3cnFvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFVFJTWUcvR1NlVDBvdHp5MU01N1phQzZXL0RYZGpaMC9SZ210aFF2c1Bja1dXSUFpQWhlTgpYdEZicWJJdHhhVXc4OUl4MDRiYkdSeHZPaW0xbnhHaERRPT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo=
Collapse
```

`/etc/rancher/rke2/rke2.yaml` on node2:

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJlVENDQVIrZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWtNU0l3SUFZRFZRUUREQmx5YTJVeUxYTmwKY25abGNpMWpZVUF4TmpNek9UVTFOemM0TUI0WERUSXhNVEF4TVRFeU16WXhPRm9YRFRNeE1UQXdPVEV5TXpZeApPRm93SkRFaU1DQUdBMVVFQXd3WmNtdGxNaTF6WlhKMlpYSXRZMkZBTVRZek16azFOVGMzT0RCWk1CTUdCeXFHClNNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJCbE1VVjJhMnV3R2pzTzR4anVyMkxsbFdwOVR4cWF0Zk0vUm5ITzUKZUNkRTd4akNrb1lUendlTjNXeXp1TWQvUnA4aVVuSnZueFNFK2FIWDVFUjh0bm1qUWpCQU1BNEdBMVVkRHdFQgovd1FFQXdJQ3BEQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01CMEdBMVVkRGdRV0JCU1RSd0tHS3NrWFcrd0I5Q1JnCnVGMms1K3hJaVRBS0JnZ3Foa2pPUFFRREFnTklBREJGQWlBdm5rSWx0L0l4eEJoS0NzcUNud0tOK3RnUXNyOTcKNHhUMTMwSkphZExPV0FJaEFLblUwOVIxSkR5RElYcnJyOE5kd0lwMnV3TmwwZ0JIZ1JVT2g3M1VVZ3lRCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
    server: https://127.0.0.1:6443
  name: default
contexts:
- context:
    cluster: default
    user: default
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: default
  user:
    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJrVENDQVRpZ0F3SUJBZ0lJSXFpMlRUeFZJL1V3Q2dZSUtvWkl6ajBFQXdJd0pERWlNQ0FHQTFVRUF3d1oKY210bE1pMWpiR2xsYm5RdFkyRkFNVFl6TXprMU5UYzNPREFlRncweU1URXdNVEV4TWpNMk1UaGFGdzB5TWpFdwpNVEV4TWpNMk1UaGFNREF4RnpBVkJnTlZCQW9URG5ONWMzUmxiVHB0WVhOMFpYSnpNUlV3RXdZRFZRUURFd3h6CmVYTjBaVzA2WVdSdGFXNHdXVEFUQmdjcWhrak9QUUlCQmdncWhrak9QUU1CQndOQ0FBUUlWRG1iZ2FtWVVEREkKbytwLytqOVNkclc5cHkrcmdLaEVjL1IzRWN1MlBXb1AyVG5abG1PWHZzQ1dLaXRoR1dGK0IxcHhPUng2Rk40eQpvYnl1aTFUbW8wZ3dSakFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJS3dZQkJRVUhBd0l3Ckh3WURWUjBqQkJnd0ZvQVVqNnRhZVhuUUtMOGpSWkJlNjR5N0F3QWY5bmN3Q2dZSUtvWkl6ajBFQXdJRFJ3QXcKUkFJZ2Z4WFNxTDBLNEQ3dFhNVTUrYVNxbHgvSmZrVUlCdzVHMEpJTkUrQXJOaTRDSURvb1JyME1pR24wbW95UwppT0VubkVZeUlrUkdpMmZSR2xyakczMTBhZ2RGCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJlakNDQVIrZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWtNU0l3SUFZRFZRUUREQmx5YTJVeUxXTnMKYVdWdWRDMWpZVUF4TmpNek9UVTFOemM0TUI0WERUSXhNVEF4TVRFeU16WXhPRm9YRFRNeE1UQXdPVEV5TXpZeApPRm93SkRFaU1DQUdBMVVFQXd3WmNtdGxNaTFqYkdsbGJuUXRZMkZBTVRZek16azFOVGMzT0RCWk1CTUdCeXFHClNNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJGRjJwUU1qN1c3MURPUmRxYmRXcE5xMTZvckVaVWx6S0lBaUdwZ2cKclZ6MkpZWk1FcGxTMlZ3cDhKQ1hRbDNKNHlSN0ZDUmVyWDNxMEZvOXNWaFkzaGVqUWpCQU1BNEdBMVVkRHdFQgovd1FFQXdJQ3BEQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01CMEdBMVVkRGdRV0JCU1BxMXA1ZWRBb3Z5TkZrRjdyCmpMc0RBQi8yZHpBS0JnZ3Foa2pPUFFRREFnTkpBREJHQWlFQXFWNjgvM0taTHEzaGRlaEx3SXY4YkVueHd6RDUKTjBqSmFsZWY0WjlHVCs4Q0lRRHYyWVVvMUIyQ2M5MFloYStxVlZFcjFMY0RvaHdZZ2pEL0tOdk5BbzgrdkE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
    client-key-data: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSUhJR3ByalBCbVlPSnpsQlB4OFplbkdiZXdGcEdRMmxlOWpBQ1l0QmloSVVvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFQ0ZRNW00R3BtRkF3eUtQcWYvby9VbmExdmFjdnE0Q29SSFAwZHhITHRqMXFEOWs1MlpaagpsNzdBbGlvcllSbGhmZ2RhY1RrY2VoVGVNcUc4cm90VTVnPT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo=
```

**Comment by FrankYang0529** [comment]
I can reproduce the bug. When I tried to use default kubeconfig on 4th node. I got error message `Unable to connect to the server: x509: certificate signed by unknown authority`.

**Comment by bk201** [comment]
@guangbochen Should an agent node (worker) has permission to access a k8s API server.
I did a rke2 installation on OpenSUSE (1 server + 1 agent) and `/etc/rancher/rke2/rke2.yaml` doesn't exist on the agent node.

The `/etc/rancher/rke2/rke2.yaml` file on the agent node[ is generated by the temporarily rke2-server when preloading images](https://github.com/harvester/harvester-installer/blob/af921bf103429b6801266862cee251aa9df3e073/package/harvester-os/files/usr/sbin/harv-install#L125-L147) (I forget to remove it).

**Comment by guangbochen** [comment]
> @guangbochen Should an agent node (worker) has permission to access a k8s API server. I did a rke2 installation on OpenSUSE (1 server + 1 agent) and `/etc/rancher/rke2/rke2.yaml` doesn't exist on the agent node.
> 
> The `/etc/rancher/rke2/rke2.yaml` file on the agent node[ is generated by the temporarily rke2-server when preloading images](https://github.com/harvester/harvester-installer/blob/af921bf103429b6801266862cee251aa9df3e073/package/harvester-os/files/usr/sbin/harv-install#L125-L147) (I forget to remove it).

It makes sense that the agent node does not have permission to access the API server, therefore removing the `rke2.yaml` file from the agent node would address the above issue, and users should be promoted to download the `KUBECONFIG` from the UI instead, thanks.

**Comment by TachunLin** [comment]
Verified **fixed** on master-d05a6cc7-head. (harvester) 
and master-14be631b-head (harvester-installer) 
Close this issue.

### Result
rke2.yaml have been removed from `/etc/rancher/rke2/` on agent node.  

Checked in the following scenarios
1. Two harvester nodes create from ipxe example at the same time
    (server node: with rke2, agent node: **w/o rke2**) 
    
2. One harvester node and add one agent node from iso installation
    (server node: with rke2, 2th agent node: **w/o rke2**) 
    
3. Three harvester nodes and add one agent node 
    (1st, 2nd, 3rd server nodes: with rke2, 4th node: **w/o rke2**)

### Test Information
* Test Environment: 3/2/1 nodes harvester on local qemu vm (as above senarios)
* Harvester version: master-d05a6cc7-head
* Harvester installer version: master-14be631b-head

### Verify Steps

1. Create 2 nodes cluster by using ipxe example
2. ssh to server and agent node 
3. check rke2.yaml in `/etc/rancher/rke2` 
4. Iso install 1 node and add the 2nd node 
5. follow step2, step3
4. Create 3 nodes cluster by using ipxe example
5. Add the 4th node 
6. follow step, step3

### Additional Context

![image](https://user-images.githubusercontent.com/29251855/139623943-7227a5f4-51e9-42da-91e3-01f3ac800b18.png)
